### PR TITLE
Bluetooth: controller: Privacy filtering in advertiser

### PR DESCRIPTION
**Pull request by carlescufi**
Implement privacy-enabled filtering in the advertiser role. This
includes all required checks when running address generation and
resolution so that the advertiser complies with the relevant
specification sections.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/ctrl.c
Comment (jhedberg) [inline subsys/bluetooth/controller/ll_sw/ctrl.c]: The whole #else and else { branches seem unnecessary since you return from within the first if-branch. I.e. the "return !memcmp(..." can be on the top level of the function as far as I can see.

File: subsys/bluetooth/controller/ll_sw/ll_filter.c
Comment (jhedberg) [inline subsys/bluetooth/controller/ll_sw/ll_filter.c]: Same thing here regarding the else { branch, however I think checkpatch also complained of this

**Review by jhedberg — DISMISSED** [review]

**Review by jhedberg — DISMISSED** [review]
Looks good enough to me, but still waiting for review by @cvinayak

**Review by jhedberg — DISMISSED** [review]

File: subsys/bluetooth/controller/ll_sw/ctrl.c
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ctrl.c]: Not necessary to have "!= 0". Could use !! .

File: subsys/bluetooth/controller/ll_sw/ctrl.c
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ctrl.c]: Combine with line 5157 code block.

File: subsys/bluetooth/controller/ll_sw/ll_filter.c
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ll_filter.c]: ASSERT before loading i.
load/assign i.
ASSERT rl[i].taken.

File: subsys/bluetooth/controller/ll_sw/ll_filter.c
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ll_filter.c]: Remove this and use ctrl_rl_enabled() in caller. Caller can then short-circuit passing the two parameters, saves some cycles.

File: subsys/bluetooth/controller/ll_sw/ctrl.c
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ctrl.c]: Use (!ctrl_rl_enabled() || ctrl_rl_allowed(...))... see my other comment in ctrl_rl_allowed definition.

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/ll_filter.c
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ll_filter.c]: ok.. lets keep it as-is for now.

File: subsys/bluetooth/controller/ll_sw/ctrl.c
Comment (carlescufi) [inline subsys/bluetooth/controller/ll_sw/ctrl.c]: Will optimize later in another way

File: subsys/bluetooth/controller/ll_sw/ll_filter.c
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ll_filter.c]: refactor the double deference. use auto-variable for &rl[i].id_addr.val[0].

**Review by cvinayak — CHANGES_REQUESTED** [review]

**Review by cvinayak — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**
